src/Components/SaveDetailsCheckbox.res

```diff
@@ -46,13 +48,21 @@ let make = (~isChecked, ~setIsChecked) => {
     ? ("Checkbox--checked", "CheckboxInput--checked", "CheckboxLabel--checked")
     : ("", "", "")
 
+  let valueToBeRendered = if showFields {
+    localeString.saveCardDetails
+  } else if business.name !== "" {
```

Comment:
use this conditional logic while setting business.name in option atom

Reply:
Updated